### PR TITLE
(prometheus) align query time range

### PR DESCRIPTION
**Pull request by mtanda**
By this changes, Prometheus query time range is always align 1 minutes when time range is longer than 1 minutes.

I use nginx as frontend of Prometheus, and cache query result for short time to reduce Prometheus load.

This PR is very useful for some situation.
The situation is "many user look same dashboard, and set auto refresh".
In such situation, too many query is executed by Prometheus simultaneously when refreshing dashboard.
(see https://github.com/grafana/grafana/pull/4220 )

The time range is set like "now-1h" - "now", so the actual query time range for Prometheus is get from Browser time, so the time range could be slightly differ.
The cache key of nginx is "query request url", these query doesn't use cache...

### CODE REVIEW

**Comment by torkelo** [comment]
> By this changes, Prometheus query time range is always align 1 minutes when time range is longer than 1 minutes.

That does not make sense. If I have a time range like last 5 minutes, I do not want it aligned to whole minutes.

**Comment by mtanda** [comment]
I mainly want to avoid "continuous too many query" by auto refresh.
So, change condition, align time when auto refresh is set.

File: public/app/plugins/datasource/prometheus/datasource.ts [hunk 1]
```diff
@@ -279,6 +283,15 @@ export function PrometheusDatasource(instanceSettings, $q, backendSrv, templateS
     return metricName + '{' + labelPart + '}';
   };
 
+  this.alignTime = function(time, range) {
```
Comment (torkelo) [inline public/app/plugins/datasource/prometheus/datasource.ts]: This alignment still does not make sense. The alignment should not cause more than a 1-3 percent change in time window. If I have last 5 minutes, and have metrics every second auto refresh to every 5sec I want live, not new data every minute

Comment (mtanda) [inline public/app/plugins/datasource/prometheus/datasource.ts]: I understand what is the problem.
Change implementation.
If refresh period is shorter than 60sec, use refresh period as align interval.

Comment (torkelo) [inline public/app/plugins/datasource/prometheus/datasource.ts]: No has nothing to do with refresh period, but time window size, The alignment should not change the time window by a large margin

**Comment by torkelo** [comment]
sorry @mtanda closing this, please open a feature request and describe the use case / scenario / reason

**Pull request closed without merging**